Chapter11/recipe-03/cxx-example/src/CMakeLists.txt

```diff
@@ -0,0 +1,196 @@
+  OUTPUT_NAME "message"
+  )
+install(
+  TARGETS
+    message-static
+  EXPORT
+    "messageTargets-static"
+  ARCHIVE DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}
+  INCLUDES DESTINATION
+    ${CMAKE_INSTALL_INCLUDEDIR}
+  )
+install(
+  EXPORT
+    "messageTargets-static"
+  NAMESPACE
+    "message::"
+  DESTINATION
+    ${CMAKECONFIG_INSTALL_DIR}
+  )
+
+add_library(message-shared SHARED $<TARGET_OBJECTS:message-objs>)
+target_link_libraries(message-shared
+  PUBLIC
+    $<$<NOT:$<BOOL:${WIN32}>>:PkgConfig::UUID>
+  )
+target_compile_definitions(message-shared
+  INTERFACE
+    $<INSTALL_INTERFACE:USING_message>
+  )
+target_include_directories(message-shared
+  INTERFACE
+    $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+  )
+set_target_properties(message-shared PROPERTIES
+  MACOSX_RPATH ON
+  SOVERSION 1
+  OUTPUT_NAME "message"
+  )
+install(
+  TARGETS
+    message-shared
+  EXPORT
+    "messageTargets-shared"
+  RUNTIME DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}
+  LIBRARY DESTINATION
+    ${CMAKE_INSTALL_LIBDIR}
+  INCLUDES DESTINATION
+    ${CMAKE_INSTALL_INCLUDEDIR}
+  )
+install(
+  EXPORT
+    "messageTargets-shared"
+  NAMESPACE
+    "message::"
+  DESTINATION
+    ${CMAKECONFIG_INSTALL_DIR}
+  )
+
+add_executable(hello-world hello-world.cpp)
+target_link_libraries(hello-world
+  PUBLIC
+    message-shared
+  )
+set_target_properties(hello-world
+  PROPERTIES
+    MACOSX_RPATH ON
+    SKIP_BUILD_RPATH OFF
+    BUILD_WITH_INSTALL_RPATH OFF
+    INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+    INSTALL_RPATH_USE_LINK_PATH ON
+  )
+install(
+  TARGETS
+    hello-world
+  EXPORT
+    "messageTargets-hello-world"
+  RUNTIME DESTINATION
+    ${CMAKE_INSTALL_BINDIR}
+  )
+install(
+  EXPORT
+    "messageTargets-hello-world"
+  NAMESPACE
+    "message::"
+  DESTINATION
+    ${CMAKECONFIG_INSTALL_DIR}
+  )
+
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(
+  ${CMAKE_CURRENT_BINARY_DIR}/messageConfigVersion.cmake
+  VERSION 1.0.0
+  COMPATIBILITY SameMajorVersion
+  )
+
+configure_package_config_file(
+  ${PROJECT_SOURCE_DIR}/cmake/messageConfig.cmake.in
+  ${CMAKE_CURRENT_BINARY_DIR}/messageConfig.cmake
+  INSTALL_DESTINATION ${CMAKECONFIG_INSTALL_DIR}
+  )
+
+install(
+  FILES
+    ${CMAKE_CURRENT_BINARY_DIR}/messageConfig.cmake
+    ${CMAKE_CURRENT_BINARY_DIR}/messageConfigVersion.cmake
+  DESTINATION
+    ${CMAKECONFIG_INSTALL_DIR}
+  )
+
+set(CPACK_PACKAGE_NAME "message")
+set(CPACK_PACKAGE_VENDOR "CMakeRecipes")
+set(CPACK_PACKAGE_DESCRIPTION_FILE "${PROJECT_SOURCE_DIR}/INSTALL.md")
+set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "message: a small messaging library")
+set(CPACK_RESOURCE_FILE_LICENSE "${PROJECT_SOURCE_DIR}/LICENSE")
+set(CPACK_PACKAGE_VERSION_MAJOR "1")
+set(CPACK_PACKAGE_VERSION_MINOR "0")
+set(CPACK_PACKAGE_VERSION_PATCH "0")
+set(CPACK_SOURCE_IGNORE_FILES "/build/;/.git/;.gitignore;/external/")
+
+set(CPACK_GENERATOR "DEB")
+set(CPACK_DEBIAN_PACKAGE_NAME "message")
```

Comment:
You don't need that because the default value for this var is CPACK_PACKAGE_NAME which is altready defined.